src/nodes.coffee

```diff
@@ -4848,25 +4895,46 @@ mergeLocationData = (locationDataA, locationDataB) ->
 # location data object that encompasses the location data of both nodes. So the
 # new `start` value will be the earlier of the two nodes’ `start` values, the
 # new `end` value will be the later of the two nodes’ `end` values, etc.
-mergeAstLocationData = (nodeA, nodeB) ->
+mergeAstLocationData = (nodeA, nodeB, {justLeading, justEnding} = {}) ->
```

Comment:
Extended `mergeAstLocationData()` to accept optional `justLeading`/`justEnding` modifiers for when we only want to consider `nodeB`'s start or end location data (ie when we only want to extend `nodeA`'s location data in a single direction rather than potentially in both directions)